### PR TITLE
Add api for sources

### PR DESCRIPTION
**Pull request by Rup-Narayan-Rajbanshi**
## Changes

- Add api for listing sources with filters

Mention related users here if any.

## This PR doesn't introduce any:

- [x] temporary files, auto-generated files or secret keys
- [x] n+1 queries
- [x] flake8 issues
- [x] `print`
- [x] typos
- [x] unwanted comments

## This PR contains valid:

- [ ] tests
- [x] permission checks (tests here too)

### CODE REVIEW

File: apps/visualization/schema.py [hunk 1]
```diff
@@ -81,6 +84,50 @@ async def get_country_profile_object(iso3):
...
+                    indicator_id=value['indicator_id'],
+                    indicator_month=value['latest_month'],
+                    subvariable=value['subvariable'],
+                    thematic=value['thematic'],
+                    topic=value['topic'],
+                    type=value['type'],
+                ) for value in data
+            ]
+        )
+        return list(DataGranular.objects.filter(filters))
```
Comment (thenav56) [inline apps/visualization/schema.py]: Can you explain what is happening here? cc: @bimal125

**Review by thenav56 — CHANGES_REQUESTED** [review]

File: apps/visualization/schema.py [hunk 2]
```diff
@@ -81,6 +84,41 @@ async def get_country_profile_object(iso3):
...
+        'indicator_id': indicator_id,
+        'indicator_name': indicator_name,
+        'indicator_description': indicator_description,
+        'subvariable': subvariable,
+        'themmatic': thematic,
+        'topic': topic,
+        'type': type,
+    })
+    qs = DataGranular.objects.filter(**filters)
+    latest_date = qs.latest('indicator_month').indicator_month
```
Comment (bimal125) [inline apps/visualization/schema.py]: Check if qs.exists()

**Review by bimal125 — CHANGES_REQUESTED** [review]

File: apps/visualization/schema.py (hunk 1 above)
Comment (Rup-Narayan-Rajbanshi) [reply apps/visualization/schema.py]: @thenav56 , Actually, we need a api to list the sources [DataGranular]. These objects should be the latest month data

**Review by bimal125 — APPROVED** [review]

File: apps/visualization/schema.py [hunk 3]
```diff
@@ -81,6 +84,43 @@ async def get_country_profile_object(iso3):
...
+        'indicator_id': indicator_id,
+        'indicator_name': indicator_name,
+        'indicator_description': indicator_description,
+        'subvariable': subvariable,
+        'thematic': thematic,
+        'topic': topic,
+        'type': type,
+    })
+    qs = DataGranular.objects.filter(**filters)
+    if qs:
```
Comment (thenav56) [inline apps/visualization/schema.py]: This is always true.

**Review by thenav56 — CHANGES_REQUESTED** [review]

File: apps/visualization/schema.py (hunk 1 above)
Comment (thenav56) [reply apps/visualization/schema.py]: New change invalidates this review comment.

File: apps/visualization/schema.py (hunk 3 above)
Comment (Rup-Narayan-Rajbanshi) [reply apps/visualization/schema.py]: @thenav56 , May be it would be fine if we use qs.exists()

Comment (thenav56) [reply apps/visualization/schema.py]: Maybe we can just use the latest_date.
```python3
....
latest_data_granular = qs.latest('indicator_month')
if not latest_data_granular:
    return []
data = qs.filter(indicator_month=indicator_month.latest_date).values(
.....
```

**Review by thenav56 — APPROVED** [review]

**Pull request merged**